CMakeLists.txt

```diff
@@ -40,6 +40,8 @@ option(HIOP_WITH_VALGRIND_TESTS "Run valgrind on certain integration tests" OFF)
 option(HIOP_BUILD_DOCUMENTATION "Build HiOp documentation via Doxygen" ON)
 
 set(HIOP_CTEST_LAUNCH_COMMAND "" CACHE STRING "Manually override launch commands used when running CTest")
+set(HIOP_CTEST_OUTPUT_DIR ${PROJECT_BINARY_DIR} CACHE PATH "Directory where CTest outputs are saved")
+mark_as_advanced(HIOP_CTEST_OUTPUT_DIR) # Hide this var as the default will be overriden rarely
```

Comment:
[nitpick] I'm not sure this needs to be advanced. IMO it just hurts discoverability, and if users are going to mess with cmake variables they don't understand, marking this var as advanced won't prevent them from doing so.